cpp/include/copc-lib/io/copc_config.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef COPCLIB_IO_COPC_CONFIG_H_
+#define COPCLIB_IO_COPC_CONFIG_H_
+
+#include "copc-lib/copc/extents.hpp"
+#include "copc-lib/copc/info.hpp"
+#include "copc-lib/geometry/vector3.hpp"
+#include "copc-lib/las/header.hpp"
+#include "copc-lib/las/vlr.hpp"
+
+namespace copc
+{
+
+// COPC Config to create new files
+struct CopcConfig
```

Comment:
name this WriterConfig or something if you approach it this way. But, I think this should be restructured to be the same object we return everywhere. So if we do writer.GetCopcConfig().NumEbItems() or writer.GetCopcConfig().GetLasHeader().SetMin({2,3,4})

Reply:
So the idea is that the object the user creates and the object we return when they want to get the header once the file is written should be one and the same.

i.e. there should be one LasHeader object and one CopcHeader object which the user passes to the writer:

```
lasconfig = new LasHeaderConfig()
lasconfig.SetPrdf(6)
writer = NamedWriter(path, lasconfig, new CopcHeader()) //lasheaderconfig is a subset of LasHeader
lasheader = writer.GetLasHeader() // returns a LasHeader
lasheader.SetPrdf(5) // big fat error!
lasheader.set_min({0,1,2})
writer.close()

reader = NamedReader(path)
lasheader = reader.GetLasHeader() // returns a LasHeader
reader.GetWkt() // this function doesn't exist!
lasheader.set_min(1,2,3); // can't do that from a reader!

```

Reply:
not really  sure what i'm going for but lmk your ideas.